### PR TITLE
tests/functional: simplify rose-stem tests

### PR DESCRIPTION
**Pull request by oliver-sanders**
* Change class scoped fixtures to module scoped. This avoids test classes and ensures fixtures are only initiated once.
* Merge fixtures which were only used once in with the test which used them.
* Remove parameterisations and fail testing code which weren't used by the tests.
* Remove monkeypatch.delenv where it wasn't needed (monkeypatch resets things by default).

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` (and `conda-environment.yml` if present).
- [x] Tests are included (or explain why tests are not needed).
- [x] `CHANGES.md` entry included if this is a change that can affect users
- [x] [Cylc-Doc](https://github.com/cylc/cylc-doc) pull request opened if required at cylc/cylc-doc/pull/XXXX.
- [x] If this is a bug fix, PR should be raised against the relevant `?.?.x` branch.

### CODE REVIEW

File: tests/functional/test_rose_stem.py [hunk 1]
```diff
@@ -251,234 +252,83 @@ def _inner_fn(rose_stem_opts, verbosity=verbosity):
     yield _inner_fn
 
 
-@pytest.fixture(scope='class')
```
Comment (oliver-sanders) [inline tests/functional/test_rose_stem.py:254]: Class-scoped fixture wasn't needed as it was only used once (it's a bit too specific to the test to be reusable).

File: tests/functional/test_rose_stem.py [hunk 2]
```diff
@@ -251,234 +252,83 @@ def _inner_fn(rose_stem_opts, verbosity=verbosity):
...
+def test_rose_stem_run_really_basic(rose_stem_run_template, setup_stem_repo):
+    """Check that assorted variables have been exported."""
     rose_stem_opts = {
         'stem_groups': [],
         'stem_sources': [
             str(setup_stem_repo['workingcopy']), "fcm:foo.x_tr@head"
         ],
     }
-    yield rose_stem_run_template(rose_stem_opts)
+    plugin_result = rose_stem_run_template(rose_stem_opts)
```
Comment (oliver-sanders) [inline tests/functional/test_rose_stem.py:263]: I'm not sure `plugin_result` is the right name for this, any suggestions?

File: tests/functional/test_rose_stem.py [hunk 3]
```diff
@@ -251,234 +252,83 @@ def _inner_fn(rose_stem_opts, verbosity=verbosity):
...
-        [
-            "run_ok",
-            "RUN_NAMES=['earl_grey', 'milk', 'sugar', 'spoon', 'cup', 'milk']",
-            "SOURCE_FOO=\"{workingcopy} fcm:foo.x_tr@head\"",
-            "HOST_SOURCE_FOO=\"{hostname}:{workingcopy} fcm:foo.x_tr@head\"",
-            "SOURCE_FOO_BASE=\"{workingcopy}\"\n",
-            "SOURCE_FOO_BASE=\"{hostname}:{workingcopy}\"\n",
-            "SOURCE_FOO_REV=\"\"\n",
-            "SOURCE_FOO_MIRROR=\"fcm:foo.xm/trunk@1\"\n",
-        ]
```
Comment (oliver-sanders) [inline tests/functional/test_rose_stem.py:296]: This parameterisation only defines on variable, `expected`, which is always `run_ok`.

I don't think the other stuff did anything?

By removing this, have I killed what this was supposed to be testing?

File: tests/functional/test_rose_stem.py [hunk 4]
```diff
@@ -75,14 +75,17 @@
...
+from _pytest.monkeypatch import MonkeyPatch
 
 from cylc.rose.stem import (
     RoseStemVersionException,
     get_rose_stem_opts,
     rose_stem,
 )
 
+
 HOST = get_host().split('.')[0]
```
Comment (oliver-sanders) [inline tests/functional/test_rose_stem.py:89]: This variable doesn't appear to be used any more?

File: tests/functional/test_rose_stem.py [hunk 5]
```diff
@@ -489,49 +339,17 @@ def with_config(
         '[rose-stem]\n'
         'automatic-options=MILK=true\n'
     )
-    monkeymodule.setenv(
+    monkeypatch.setenv(
         'ROSE_CONF_PATH', str(setup_stem_repo['basetemp'])
     )
-    yield rose_stem_run_template(rose_stem_opts)
-    monkeymodule.delenv('ROSE_CONF_PATH')
```
Comment (oliver-sanders) [inline tests/functional/test_rose_stem.py:496]: Monkeypatch/module should do this by itself?

**Comment by oliver-sanders** [comment]
Test failures are now all reinstall tests.

File: tests/functional/test_rose_stem.py [hunk 6]
```diff
@@ -75,14 +75,17 @@
 from cylc.flow.hostuserutil import get_host
 from cylc.flow.pathutil import get_workflow_run_dir
 from metomi.rose.resource import ResourceLocator
+
 import pytest
+from _pytest.monkeypatch import MonkeyPatch
```
Comment (MetRonnie) [inline tests/functional/test_rose_stem.py:80]:
```suggestion
from pytest import MonkeyPatch
```

File: tests/functional/test_rose_stem.py (hunk 3 above)
Comment (oliver-sanders) [reply tests/functional/test_rose_stem.py:296]: Yes, I had misread that parametrisation, drafted.

File: tests/functional/test_rose_stem.py (hunk 1 above)
Comment (wxtim) [reply tests/functional/test_rose_stem.py:254]: Much of the class scoped stuff was done in the name of doing as little re-writing as possible when moving the tests from Stuarts work.

File: tests/functional/test_rose_stem.py (hunk 2 above)
Comment (wxtim) [reply tests/functional/test_rose_stem.py:263]: * stem_outputs
* stem_variables

File: tests/functional/test_rose_stem.py (hunk 5 above)
Comment (wxtim) [reply tests/functional/test_rose_stem.py:496]: At the end of the module. I think we want to remove this now.

**Comment by oliver-sanders** [comment]
Superseded by #314

**Pull request closed without merging**